src/parser.py

```diff
@@ -1,158 +1,370 @@
 import argparse
 import sys
 from os import getcwd
+from shlex import split
+from app import *
+
+class ParserTemplate:
+
+	'''This class contains a parser that has includes commands and arguments that are common between the standalone terminal commands and the REPL'''
+
+	def __init__(self):
+		#general parser
+		self.parser = argparse.ArgumentParser(prog = "hashesdb", description = "Manage database that contains hash values.")
+		self.subparsers = self.parser.add_subparsers(description = "To print detailed help about a specific subcommand, use the -h option. For example: hashesdb search -h", help = "subcommand description")
+		self.parser.add_argument('-v','--version', action = 'version', version = '%(prog)s 1.0', help = "print the program's version")
+
+		#about subcommand parser
+		about_help_msg = "print information about the program"
+		self.parser_about = self.subparsers.add_parser('about', help = about_help_msg, description = about_help_msg)
+
+		#schema subcommand parser
+		schema_help_msg = "print schema documentation"
+		self.parser_schema = self.subparsers.add_parser('schema', help = schema_help_msg, description = schema_help_msg)	
+
+		#create subcommand parser
+		create_help_msg = "create a hashesdb database at a gien path"
+		self.parser_create = self.subparsers.add_parser('create', help= create_help_msg, description = create_help_msg)
+		self.parser_create.add_argument('-d', '--database', '--db', required = True, metavar = "DATABASE_PATH", action = "store", help = "path to the new hashesdb database (.db file)")
```

Comment:
since this is common to all(?) subcommands, why not have it in on the parent `parser` ?

Reply:
This is not common to all subcommands since commands like `about` and `schema` do not need a `-d` argument.